src/vs/base/browser/ui/selectBox/selectBox.ts

```diff
@@ -152,4 +426,243 @@ export class SelectBox extends Widget {
+		// Need to be visible to measure
+
+		dom.toggleClass(this.selectDropDownContainer, 'visible', true);
+
+		const selectWidth = dom.getTotalWidth(this.selectElement);
+		const selectPosition = dom.getDomNodePagePosition(this.selectElement);
+
+		// Set container height to max from select bottom to margin above status bar
+		const statusBarHeight = dom.getTotalHeight(document.getElementById('workbench.parts.statusbar'));
+		const maxSelectDropDownHeight = (window.innerHeight - selectPosition.top - selectPosition.height - statusBarHeight - SelectBox.SELECT_DROPDOWN_BOTTOM_MARGIN);
+
+		// SetUp list dimensions and layout - account for container padding
+		if (this.selectList) {
+			this.selectList.layout();
+			let listHeight = this.selectList.contentHeight;
+			const listContainerHeight = dom.getTotalHeight(this.selectDropDownListContainer);
+			const totalVerticalListPadding = listContainerHeight - listHeight;
+
+			// Always show complete list items - never more than Max available vertical height
+			if (listContainerHeight > maxSelectDropDownHeight) {
+				listHeight = ((Math.floor((maxSelectDropDownHeight - totalVerticalListPadding) / this.getHeight())) * this.getHeight());
+			}
+
+			// Use parent select font
+			this.cloneElementFont(this.selectElement, this.selectDropDownContainer);
+			this.selectList.layout(listHeight);
+			this.selectList.domFocus();
+
+			// Finally set focus on selected item
+			this.selectList.setFocus([this.selected]);
+			this.selectList.reveal(this.selectList.getFocus()[0]);
+
+			// Set final container height after adjustments
+			this.selectDropDownContainer.style.height = (listHeight + totalVerticalListPadding) + 'px';
+
+			// Determine optimal width - min(longest option), opt(parent select), max(ContextView controlled)
+			const selectMinWidth = this.setWidthControlElement(this.widthControlElement);
+			const selectOptimalWidth = Math.max(selectMinWidth, Math.round(selectWidth)).toString() + 'px';
+
+			this.selectDropDownContainer.style.minWidth = selectOptimalWidth;
+
+			// Maintain focus outline on parent select as well as list container - tabindex for focus
+			this.selectDropDownListContainer.setAttribute('tabindex', '0');
+			dom.toggleClass(this.selectElement, 'synthetic-focus', true);
+			dom.toggleClass(this.selectDropDownContainer, 'synthetic-focus', true);
+		}
+	}
+
+	private setWidthControlElement(container: HTMLElement): number {
+		let elementWidth = 0;
+
+		if (container && !!this.options) {
+			let longest = 0;
+
+			for (var index = 0; index < this.options.length; index++) {
+				if (this.options[index].length > this.options[longest].length) {
+					longest = index;
+				}
+			};
+
+			container.innerHTML = this.options[longest];
+			this.cloneElementFont(this.selectElement, container);
+			elementWidth = dom.getTotalWidth(container);
+		}
+
+		return elementWidth;
+	}
+
+	public cloneElementFont(source: HTMLElement, target: HTMLElement) {
```

Comment:
@cleidigh not so happy:
* why is the method `public`?
* it uses `var`, which we try to avoid (e.g. use `let` or `const` instead)
* can the value be cached and reused instead of always asking for it?